packages/cardano-services/src/Program/programs/blockfrostWorker.ts

```diff
@@ -0,0 +1,71 @@
+import { BlockfrostWorker, BlockfrostWorkerConfig, getPool } from '../services';
+import { CommonProgramOptions, PosgresProgramOptions, PostgresOptionDescriptions } from '../options';
+import { Logger } from 'ts-log';
+import { MissingProgramOption } from '..';
+import { SrvRecord } from 'dns';
+import { createDnsResolver } from '../utils';
+import { createLogger } from 'bunyan';
+
+export const BLOCKFROST_WORKER_API_URL_DEFAULT = new URL('http://localhost:3000');
+export const CACHE_TTL_DEFAULT = 24 * 60; // One day
+export const CREATE_SCHEMA_DEFAULT = false;
+export const DROP_SCHEMA_DEFAULT = false;
+export const DRY_RUN_DEFAULT = false;
+export const SCAN_INTERVAL_DEFAULT = 60; // One hour
+
+export const availableNetworks = ['mainnet', 'preprod', 'preview'] as const;
+export type AvailableNetworks = typeof availableNetworks[number];
+
+export enum BlockfrostWorkerOptionDescriptions {
+  BlockfrostApiKey = 'Blockfrost API Key',
+  CacheTTL = 'TTL of blockfrost cached metrics in minutes',
+  CreateSchema = 'create the schema; usefull for development',
+  DropSchema = 'drop the schema; usefull for development',
+  DryRun = 'dry run; usefull for tests',
+  Network = 'network to run against',
+  ScanInterval = 'interval between a scan and the next one in minutes'
+}
+
+export type BlockfrostWorkerArgs = CommonProgramOptions & PosgresProgramOptions & BlockfrostWorkerConfig;
+
+export interface LoadBlockfrostWorkerDependencies {
+  dnsResolver?: (serviceName: string) => Promise<SrvRecord>;
+  logger?: Logger;
+}
+
+const blockfrostWorker = 'Blockfrost worker';
+
+export const loadAndStartBlockfrostWorker = async (
+  args: BlockfrostWorkerArgs,
+  deps: LoadBlockfrostWorkerDependencies = {}
+) => {
+  const logger = deps?.logger || createLogger({ level: args.loggerMinSeverity, name: 'blockfrost-worker' });
+  const dnsResolver =
+    deps?.dnsResolver ||
+    createDnsResolver(
+      {
+        factor: args.serviceDiscoveryBackoffFactor,
+        maxRetryTime: args.serviceDiscoveryTimeout
+      },
+      logger
+    );
+  const db = await getPool(dnsResolver, logger, args);
+
+  if (!args.blockfrostApiKey)
+    throw new MissingProgramOption(blockfrostWorker, BlockfrostWorkerOptionDescriptions.BlockfrostApiKey);
+
+  if (!args.network) throw new MissingProgramOption(blockfrostWorker, BlockfrostWorkerOptionDescriptions.Network);
+
+  if (!db)
+    throw new MissingProgramOption(blockfrostWorker, [
+      PostgresOptionDescriptions.ConnectionString,
+      PostgresOptionDescriptions.ServiceDiscoveryArgs
+    ]);
+
+  const blockfrost = new BlockfrostWorker(args, { db, logger });
+
+  await blockfrost.initialize();
```

Comment:
```suggestion
```

Reply:
As far as I remember calling `start` on a `RunnableModule` without calling `initialize` throws.

Reply:
Right, but the lifecycle will be handled by the `HttpService`